### PR TITLE
Cache supports nodes to determine if restore dialog needs to be shown.

### PR DESCRIPTION
**Pull request by pranavkm**
Fixes https://github.com/NuGet/Home/issues/937

cc @emgarten  @deepakaravindr @yishaigalatzer

### CODE REVIEW

File: src/PackageManagement/BuildIntegration/BuildIntegratedRestoreUtility.cs [hunk 1]
```diff
@@ -165,7 +166,10 @@ public static IReadOnlyList<PackageIdentity> GetRemovedPackages(LockFile origina
                 var closure = await project.GetProjectReferenceClosureAsync();
                 var files = closure.Select(reference => reference.PackageSpecPath).ToList();
 
-                var projectInfo = new BuildIntegratedProjectCacheEntry(project.JsonConfigPath, files);
+                var projectInfo = new BuildIntegratedProjectCacheEntry(
+                    project.JsonConfigPath,
+                    files,
+                    project.PackageSpec.RuntimeGraph.Supports.Keys.OrderBy(p => p, StringComparer.Ordinal).ToArray());
```
Comment (yishaigalatzer) [inline src/PackageManagement/BuildIntegration/BuildIntegratedRestoreUtility.cs:172]: are you guaranteed to be able to dot through supports/keys ?

Comment (pranavkm) [inline src/PackageManagement/BuildIntegration/BuildIntegratedRestoreUtility.cs:172]: Based on https://github.com/NuGet/NuGet.PackageManagement/pull/31/files#diff-def53c1d38f681d1d920797beb2d7b8fL233 `RuntimeGraph.Supports` is never null. We always create an empty one.

File: test/PackageManagement.Test/BuildIntegratedRestoreUtilityTests.cs [hunk 2]
```diff
@@ -408,6 +356,81 @@ public class BuildIntegratedRestoreUtilityTests
         }
 
         [Fact]
+        public async Task CacheHasChanges_ReturnsTrue_IfSupportProfilesDiffer()
```
Comment (deepakaravindr) [inline test/PackageManagement.Test/BuildIntegratedRestoreUtilityTests.cs:359]: For naming consistency in the file, could you rename to BuildIntegratedRestoreUtility_CacheDiffersOnSupportProfiles?

File: src/PackageManagement/BuildIntegration/BuildIntegratedRestoreUtility.cs (hunk 1 above)
Comment (deepakaravindr) [inline src/PackageManagement/BuildIntegration/BuildIntegratedRestoreUtility.cs:172]: Curious, why is it not OrdinalIgnoreCase?

**Comment by deepakaravindr** [comment]
Looks good to me

File: src/PackageManagement/BuildIntegration/BuildIntegratedRestoreUtility.cs (hunk 1 above)
Comment (pranavkm) [inline src/PackageManagement/BuildIntegration/BuildIntegratedRestoreUtility.cs:172]: The intent was if there was any change to the node we should re-restore. Changing the case seems like it.

File: test/PackageManagement.Test/BuildIntegratedRestoreUtilityTests.cs (hunk 2 above)
Comment (pranavkm) [inline test/PackageManagement.Test/BuildIntegratedRestoreUtilityTests.cs:359]: It's less meaningful though - the cache differs all the time - we just choose to return true if specific aspects of it are different. I'll leave this as is.

**Comment by pranavkm** [comment]
Updated

**Comment by emgarten** [comment]
:shipit:

**Comment by yishaigalatzer** [comment]
Please look at the report from pilcie before resolving the issue

File: src/PackageManagement/BuildIntegration/BuildIntegratedRestoreUtility.cs (hunk 1 above)
Comment (lalaithion) [reply src/PackageManagement/BuildIntegration/BuildIntegratedRestoreUtility.cs:172]: Good catch — changed `resourceListCacheLocation` from `string` to `names.CacheLocation`, parsing it once in `NewDispatcherStore` (matching the pattern in `membership/store_dispatcher.go`). This avoids re-parsing on every cache access.

**Pull request merged**